src/datasets/features.py

```diff
@@ -33,18 +34,53 @@
 
 
 def string_to_arrow(type_str: str) -> pa.DataType:
-    if type_str not in pa.__dict__:
+    """
+    string_to_arrow takes a datasets string dtype and converts it to a pyarrow.DataType.
+
+    In effect, `dt == string_to_arrow(arrow_to_datasets_dtype(dt))`
```

Comment:
```suggestion
    In effect, `dt == string_to_arrow(str(dt))`
```
`arrow_to_datasets_dtype` doesn't exist yet ^^